File: ruby/02/15.rb
# -*- coding: utf-8 -*-

=begin
15. 末尾のN行を出力
自然数Nをコマンドライン引数などの手段で受け取り，入力のうち末尾のN行だけを表示せよ．確認にはtailコマンドを用いよ．
=end

n = ARGV.shift.to_i
arr = $<.to_a
n.times do |i|
  puts arr[arr.length - n + i]
end

File: python/02/15.py
# -*- coding: utf-8 -*-
# 15. 末尾のN行を出力
# 自然数Nをコマンドライン引数などの手段で受け取り，入力のうち末尾のN行だけを表示せよ．確認にはtailコマンドを用いよ．

import sys

with open(sys.argv[2]) as f:
    lines = f.readlines()

for line in lines[len(lines) - int(sys.argv[1]):]:
    print line,
